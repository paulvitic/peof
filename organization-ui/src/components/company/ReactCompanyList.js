import React from 'react'
import PropTypes from 'prop-types';
import { StyledButton } from "./style";

const ReactCompanyList = (props) => {
    return (
        <StyledButton {...props} onClick={props.onClick}>{props.children}</StyledButton>
    );
};

ReactCompanyList.propTypes = {
    primary:PropTypes.bool,
    confirm:PropTypes.bool,
    alert:PropTypes.bool,
    disabled:PropTypes.bool.isRequired,
    onClick:PropTypes.func.isRequired
};


ReactCompanyList.defaultProps = {
    disabled:false,
    onClick:()=>{}
};

export default ReactCompanyList;
