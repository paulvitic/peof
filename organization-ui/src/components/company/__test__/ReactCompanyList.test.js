import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ReactCompanyList  from '../ReactCompanyList';

describe('App', () => {
  it('renders without crashing given the required props', () => {
    const props = {
      primary:true,
      confirm:true,
      alert:true,
      disabled:false,
      onClick:()=>{}
    };
    const wrapper = shallow(<ReactCompanyList {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});