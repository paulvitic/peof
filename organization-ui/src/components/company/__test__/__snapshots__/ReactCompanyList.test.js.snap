// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing given the required props 1`] = `
<styled.button
  alert={true}
  confirm={true}
  disabled={false}
  onClick={[Function]}
  primary={true}
/>
`;
